creating_packages/package_repo.rst

```diff
@@ -108,44 +108,57 @@ When you export the recipe (or when :command:`conan create` is called) the expor
 remote and commit of the local repository:
 
 .. code-block:: python
-   :emphasize-lines: 7, 8
+   :emphasize-lines: 8-10
 
+    import os
     from conans import ConanFile, CMake, tools
 
     class HelloConan(ConanFile):
          scm = {
             "type": "git",  # Use "type": "svn", if local repo is managed using SVN
             "subfolder": "hello",
             "url": "auto",
-            "revision": "auto"
+            "revision": "auto",
+            "password": os.environ.get("SECRET", None)
          }
         ...
 
 You can commit and push the *conanfile.py* to your origin repository, which will always preserve the ``auto``
-values. But when the file is exported to the Conan local cache (except you have uncommitted changes, read below),
-the copied recipe in the local cache will point to the captured remote and commit:
-
-.. code-block:: python
-   :emphasize-lines: 7, 8
-
-    from conans import ConanFile, CMake, tools
-
-    class HelloConan(ConanFile):
-         scm = {
-            "type": "git",
-            "subfolder": "hello",
-            "url": "https://github.com/conan-io/hello.git",
-            "revision": "437676e15da7090a1368255097f51b1a470905a0"
-         }
-        ...
-
-So when you :ref:`upload the recipe <uploading_packages>` to a Conan remote, the recipe will contain
-the "resolved" URL and commit.
-
-When you are requiring your ``HelloConan``, the :command:`conan install` will retrieve the recipe from the
+values. When the file is exported to the Conan local cache (except you have uncommitted changes, read below),
+these data will be stored in the *conanfile.py* itself (Conan will modify the file) or in a special file
+:ref:`conandata_yml` that will be stored together with the recipe, depending on the value of the configuration
+parameter :ref:`scm_to_conandata<conan_conf>`.
+
+ * If the ``scm_to_conandata`` is not activated (default behavior in Conan v1.x) Conan will store a modified
+   version of the *conanfile.py* with the values of the fields in plain text:
+
+   .. code-block:: python
+    :emphasize-lines: 8-10
+
+        import os
+        from conans import ConanFile, CMake, tools
+
+        class HelloConan(ConanFile):
+            scm = {
+                "type": "git",
+                "subfolder": "hello",
+                "url": "https://github.com/conan-io/hello.git",
+                "revision": "437676e15da7090a1368255097f51b1a470905a0",
+                "password": "MY_SECRET"
+            }
+            ...
+
+   So when you :ref:`upload the recipe <uploading_packages>` to a Conan remote, the recipe will contain
+   the "resolved" URL and commit.
+
+ * If ``scm_to_conandata`` is activated, the value of these fields (except ``username`` and ``password``) will
+   be stored in the ``conandata.yml`` file that will be automatically exported with the recipe.
+
+Whatever the option activated, the data resolved will be asigned by Conan to the corresponding field, so
```

Comment:
This sentence sounds strange to me. Better use ``Whichever *configuration* is set...`` or completely rephrase to something like:
```suggestion
The data resolved from the `scm` attribute, will be assigned by Conan to the corresponding field in relation to the `scm_to_conandata` configuration. that way, 
```